src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -87,7 +87,7 @@ public KubernetesLauncher() {
     }
 
     @Override
-    public boolean isLaunchSupported() {
+    public synchronized boolean isLaunchSupported() {
```

Comment:
I found a problem here: if this method is called from another thread used synchronously during launch (`JnlpConnectionState.fireAfterChannel`) while `launch(…)` is running, you can get a deadlock.